transport-native-unix-common/src/main/java/io/netty/channel/unix/NativeInetAddress.java

```diff
@@ -103,9 +104,6 @@ public static InetSocketAddress address(byte[] addr, int offset, int len) {
     }
 
     static int decodeInt(byte[] addr, int index) {
-        return  (addr[index]     & 0xff) << 24 |
-                (addr[index + 1] & 0xff) << 16 |
-                (addr[index + 2] & 0xff) <<  8 |
-                addr[index + 3] & 0xff;
+        return ByteBuffer.wrap(addr, index, 4).getInt();
```

Comment:
I would suggest to just check the endianess and then decode via the array directly to reduce GC

Reply:
Basically:

```java
if (PlatformDependent.BIG_ENDIAN_NATIVE_ORDER) {
            return (addr[index] & 0xff) << 24 |
                    (addr[index + 1] & 0xff) << 16 |
                    (addr[index + 2] & 0xff) << 8 |
                    addr[index + 3] & 0xff;
        } else {
            return addr[index]      & 0xff        |
                    (addr[index + 1] & 0xff) << 8  |
                    (addr[index + 2] & 0xff) << 16 |
                    (addr[index + 3] & 0xff) << 24;
        }
```

Reply:
@inatatsu ^^

Reply:
@normanmaurer Thank you very much for your suggestion. I realized we do not have to touch `decodeInt()` since the existing implementation is already neutral from the platform native endian order. I'll fix my patch to preserve the current `decodeInt()`.

Reply:
@inatatsu let me know when I should check again

Reply:
@normanmaurer Updated the patch. Please check again.